api/src/main/java/jakarta/servlet/ServletOutputStream.java

```diff
@@ -283,8 +292,17 @@ public void println(double d) throws IOException {
 
     /**
      * Instructs the <code>ServletOutputStream</code> to invoke the provided {@link WriteListener} when it is possible to
-     * write
-     *
+     * write.
+     * <p>
+     * Note that after this method has been called methods on this stream that are documented to throw {@link IOException}
+     * will no longer throw these exceptions directly, instead any exception that occurs will be reported via
+     * {@link WriteListener#onError(Throwable)}. Please refer to this method for more information. This only applies to
+     * {@code IOException}, other exception types may still be thrown (e.g. methods can throw {@link IllegalStateException}
+     * if {@link #isReady()} has not returned true).
+     * <p>
+     * Once this method has been called {@link #flush()} and {@link #close()} become asynchronous operations, they will be
+     * performed in the background and any problems will be reported through the {@link WriteListener#onError(Throwable)}
+     * method.
```

Comment:
```suggestion
     * Note that after this method has been called, methods on this stream that are documented 
     * to throw {@link IOException} might not throw these exceptions directly, 
     * instead they may be reported via {@link ReadListener#onError(Throwable)} after a call to 
     * {@link #isReady()} has returned {@code false}. 
     * Please refer to {@link #isReady()} method for more information. 
     * <p>
     * Once this method has been called {@link #flush()} and {@link #close()} become asynchronous 
     * operations, possibly performed in the background. Thus any problems may be reported through the 
     * {@link WriteListener#onError(Throwable)} method, which may be called at any time 
     * after a {@link #close()} or otherwise only after a call to {@link #isReady()} has returned
     * {@code false}.
```